internal/cluster/schema.go

```diff
@@ -115,53 +117,62 @@ func clusterResourceAttributes() map[string]resourceschema.Attribute {
 			},
 		},
 
-		// "maintenance": resourceschema.SingleNestedAttribute{
-		// 	// FIXME: if bug fixed, make Required
-		// 	// https://github.com/metal-stack-cloud/terraform-provider-metal/issues/51
-		// 	Computed:            true,
-		// 	Optional:            true,
-		// 	MarkdownDescription: "maintenance options",
-		// 	PlanModifiers: []planmodifier.Object{
-		// 		objectplanmodifier.UseStateForUnknown(),
-		// 	},
-		// 	Attributes: map[string]resourceschema.Attribute{
-		// 		"kubernetes_autoupdate": resourceschema.BoolAttribute{
-		// 			Computed:            true,
-		// 			MarkdownDescription: "Wether kubernetes autoupdate is enabled",
-		// 		},
-		// 		"machineimage_autoupdate": resourceschema.BoolAttribute{
-		// 			Computed:            true,
-		// 			MarkdownDescription: "Wether maschine image autoupdate is enabled",
-		// 		},
-		// 		"time_window": resourceschema.SingleNestedAttribute{
-		// 			// FIXME: if bug fixed, make Required
-		// 			// https://github.com/metal-stack-cloud/terraform-provider-metal/issues/51
-		// 			Computed:            true,
-		// 			MarkdownDescription: "Set time window for maintenance",
-		// 			PlanModifiers: []planmodifier.Object{
-		// 				objectplanmodifier.UseStateForUnknown(),
-		// 			},
-		// 			Attributes: map[string]resourceschema.Attribute{
-		// 				"begin": resourceschema.StringAttribute{
-		// 					Computed:            true,
-		// 					Optional:            true,
-		// 					MarkdownDescription: "Set begin of maintenance window. Use the format 'HH:MM AM/PM' and consider the UTC offset.",
-		// 					Validators: []validator.String{
-		// 						stringvalidator.RegexMatches(
-		// 							regexp.MustCompile(`^\d\d:\d\d (AM|PM)$`),
-		// 							"not a valid time of day",
-		// 						),
-		// 					},
-		// 				},
-		// 				"duration": resourceschema.Int64Attribute{
-		// 					Computed:            true,
-		// 					Optional:            true,
-		// 					MarkdownDescription: "Set duration of maintenance window. The duration must be defined in hours.",
-		// 				},
-		// 			},
-		// 		},
-		// 	},
-		// },
+		"maintenance": resourceschema.SingleNestedAttribute{
+			// Computed: true,
+			// Optional: true,
+			Required:            true,
```

Comment:
Maybe, it would be better to make this `optional`. However, the `maintenance` attribute in the cluster struct need to handle null values.